### ISSUE TITLE
no-dupe-keys doesn't work on JSX

### ISSUE THREAD

**Issue by Zacqary** [eslint/eslint#6140]
**What version of ESLint are you using?**
2.8.0
**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESlint
**Please show your full configuration:**
```javascript
{

  "parser": "babel-eslint",
  "env": {
    "browser": true,
    "es6": true
  },

  "plugins": [
    "react"
  ],
  "extends": "eslint:recommended",
  "rules": {
    "camelcase": [1, { 'properties': 'never' }],
    "consistent-return": 0,
    "indent": [2, 4, {"SwitchCase": 1}],
    "react/jsx-uses-vars": 1,
    "new-cap": 0,
    "no-console": 0,
    "no-eval": 2,
    "no-multi-spaces": 1,
    "no-bitwise": 2,
    "no-redeclare": 1,
    "no-plusplus": 0,
    "no-new": 2,
    "no-undef": 2,
    "no-underscore-dangle": 0,
    "no-unused-vars": 1,
    "no-use-before-define": 0,
    "one-var": [1, "never"],
    "quotes": 0,
    "semi": 1,
    "strict": 0
  },
  "settings": {
    "react": {
      "pragma": "React",
      "version": "0.14.6"
    }
  }
}
```
**What did you do? Please include the actual source code causing the issue.**
```javascript
<li onMouseMove={() => setItem(i)}
    onMouseMove={confirmMouseMove(() => setItem(i))}/>
```
**What did you expect to happen?**
The duplicate `onMouseMove` key should have been marked as a violation of `no-dupe-keys`

**What actually happened? Please include the actual, raw output from ESLint.**
Nothing

---

Alternatively, this could be a new JSX-specific rule.

**Comment by ilyavolodin** [comment]
`no-dupe-key` was design to work on objects. Since JSX element is not an object, this rule wouldn't trigger. I think it's a good rule to add, but I don't think it should be in the core. I would suggest requesting a new rule from `eslint-plugin-react` (since they already have a bunch of JSX-specific rules)

**Comment by Zacqary** [comment]
Thanks, I'll take it up with that project!
